partner_vat_unique/models/res_partner.py

```diff
@@ -17,10 +16,11 @@ def _check_vat_unique(self):
         for record in self:
             if record.parent_id or not record.vat:
                 continue
-            test_condition = config["test_enable"] and not self.env.context.get(
-                "test_vat"
-            )
-            if test_condition:
+            if (
```

Comment:
Put it outside the loop.  If deactivated, don't enter in it at all.

Reply:
Done!